hwmon.c

```diff
@@ -69,3 +70,66 @@ const struct hwmon_ops tt_hwmon_ops = {
 	.read_string = tt_hwmon_read_string,
 };
 
+ssize_t tt_show_attribute(struct device *dev, struct device_attribute *attr, char *buf)
+{
+	struct tenstorrent_device *tt_dev = dev_get_drvdata(dev);
+	struct tt_attribute_data *data = container_of(attr, struct tt_attribute_data, attr);
+	struct tt_hwmon_context *ctx = &tt_dev->hwmon_context;
+	u32 value = ioread32(ctx->telemetry_base + data->reg_offset);
+	value &= data->mask;
+	return sprintf(buf, "%u\n", value);
+}
+
+ssize_t tt_show_card_type(struct device *dev, struct device_attribute *attr, char *buf) {
+	struct tenstorrent_device *tt_dev = dev_get_drvdata(dev);
+	struct tt_attribute_data *data = container_of(attr, struct tt_attribute_data, attr);
+	struct tt_hwmon_context *ctx = &tt_dev->hwmon_context;
+
+	u32 board_id_hi = ioread32(ctx->telemetry_base + data->reg_offset);
+	u64 card_type = (board_id_hi >> 4) & 0xFFFF;
```

Comment:
Why is card_type u64? Seems it can't be more than 16 bits.

Reply:
Fixed